bip-musig2-descriptors.mediawiki

```diff
@@ -0,0 +1,69 @@
+<pre>
+  BIP: musig2-descriptors
+  Layer: Applications
+  Title: musig() Descriptor Key Expression
+  Author: Ava Chow <me@achow101.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-musig2-descriptor
+  Status: Draft
+  Type: Informational
+  Created: 2024-01-15
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This document specifies a <tt>musig()</tt> key expression for output script descriptors.
+<tt>musig()</tt> expressions take multiple keys and produce an aggregate public key using BIP 327.
+
+==Copyright==
+
+This BIP is licensed under the BSD 2-clause license.
+
+==Motivation==
+
+BIP 327 introduces the MuSig2 Multi-Signature scheme. It is useful to have a way for keys to be used
+in a MuSig2 aggregate key to be expressed in descriptors so that wallets can more easily use MuSig2.
+
+==Specification==
+
+A new key expression is defined: <tt>musig()</tt>.
+
+===<tt>musig()</tt>===
+
+The <tt>musig(KEY, KEY, ..., KEY)</tt> expression can only be used inside of a <tt>tr()</tt>
+expression as a key expression. The aggregate public key is produced by using the <tt>KeyAgg</tt>
+algorithm on all KEYs specified in the expression after performing all specified derivation. As with
+script expressions, KEY can contain child derivation specified by <tt>/*</tt>. A new aggregate
+public key will be computed for each child index. Keys must be sorted with the <tt>KeySort</tt>
+algorithm after all derivation and prior to aggregation.
+
+<tt>musig(KEY, KEY, ..., KEY)/NUM/.../*</tt> expressions are also allowed. The aggregate public key
+is first computed as described above, with the keys also being sorted after all derivation and prior
+to aggreation, then further BIP 32 derivation will be performed on it as described in
+[[bip-musig2-derivation.mediawiki|BIP-musig2-derivation]]. As there is no aggregate private key,
+only unhardened derivation from the aggregate public key is allowed, and thus the derivation steps
+following the <tt>musig()</tt> expression cannot contain
+<tt>/NUMh</tt> or <tt>/NUM'</tt> derivation steps nor <tt>/*h</tt>, or <tt>/*'</tt> child derivation.
+For these <tt>musig()</tt> expressions, the KEY expressions contained within must be xpubs or derived from
+xpubs, and cannot contain child derivation as specified by a <tt>/*</tt>, <tt>/*'</tt>, or <tt>/*h</tt>.
+
```

Comment:
It might be worth mentioning explicitly that `musig()` nesting is forbidden (while `musig()` inside Script-based multisignatures/thresholds is fine).

Reply:
I thought it was sufficiently implied by the above paragraph which states "The `musig(KEY, KEY, ..., KEY)` expression can only be used inside of a `tr()` expression as a key expression."

Reply:
Since you can put `musig()` inside any fragment inside `tr()`, I think it's very easy to miss the fact that `musig()` itself is not a fragment, despite using a visually identical syntax. Especially since nesting `musig()` inside `musig()` is a natural thing that some people will surely want to do.

Reply:
Added a sentence to make it explicit.